src/lib.rs

```diff
@@ -2944,13 +2944,7 @@ impl Build {
                             let cc = if target.abi == "llvm" { clang } else { gnu };
                             format!("{prefix}-{cc}").into()
                         }
-                        None => {
-                            if raw_target == "xtensa-esp32s3-espidf" {
-                                "xtensa-esp32s3-elf".into()
-                            } else {
-                                default.into()
-                            }
-                        }
+                        None => default.into(),
```

Comment:
We could keep this as a fallback for these targets, by using `self.which(cross_compile_cc, None).is_some()` to check for existence of the new compiler

Reply:
Perhaps. I thought it would be better to group all of these in one place. Also `xtensa-esp32s3-espidf` is covered in below function